pkg/helper/helpers.go

```diff
@@ -0,0 +1,37 @@
+package helper
+
+import (
+	"fmt"
+	"strings"
+
+	hypdeployment "github.com/stolostron/hypershift-deployment-controller/api/v1alpha1"
+)
+
+//TODO @ianzhang366 integrate with the clusterSet logic
+func GetTargetManagedCluster(hyd *hypdeployment.HypershiftDeployment) string {
+	if len(hyd.Spec.TargetManagedCluster) == 0 {
+		return hyd.GetNamespace()
+	}
+
+	return hyd.Spec.TargetManagedCluster
+}
+
+func GetTargetNamespace(hyd *hypdeployment.HypershiftDeployment) string {
+	if len(hyd.Spec.TargetNamespace) == 0 {
+		return hyd.GetNamespace()
+	}
+
+	return hyd.Spec.TargetNamespace
+}
+
+func ManagedClusterName(hyd *hypdeployment.HypershiftDeployment) string {
+	if strings.HasPrefix(hyd.Spec.InfraID, hyd.GetName()) {
+		return hyd.Spec.InfraID
```

Comment:
A little weird, the infraID(including 5 random chars) is contained in the managed cluster name. I am considering using `fmt.Sprintf("%s-%s", hyd.GetNamespace(), hyd.GetName())` as the managed cluster name.

Reply:
I think its going to be weird if the managedCluster name doesn't match the cluster name you choose. But I understand the problem you are trying to avoid...

Reply:
I would say using infraID would be the best option.  I don't think there will be that many overlap in names, so infraID is probably okay.

We use it in the context of a Management Cluster, but that is limited to 400 clusters.  Across the Hub, it 2k-4k names... But I still think that's okay.

Thoughts @zhujian7 @ianzhang366 ??

Reply:
Thanks. will use infraID